addons/calendar/models/calendar_event.py

```diff
@@ -437,14 +438,17 @@ def split_privacy(events):
             public = [event for event in events if event.get('privacy') != 'private']
             return private, public
 
-        def my_events(events):
+        def split_visibility(events):
             """
             :param events: list of event values (dict)
             :return: tuple(my events, other events)
             """
-            my = [event for event in events if event.get('user_id') and event.get('user_id')[0] == self.env.uid]
-            others = [event for event in events if not event.get('user_id') or event.get('user_id')[0] != self.env.uid]
-            return my, others
+            current_partner_id = self.env.user.partner_id.id
+            visible_events = [event for event in events if (event.get('user_id') and event.get('user_id')[0] == self.env.uid)
+                              or current_partner_id in event.get('partner_ids')]
+            my_ids = map(lambda e: e['id'], visible_events)
+            other_events = [event for event in events if event.get('id') not in my_ids]
```

Comment:
Not going to work, `map` is lazy and not resumable so `in` is going to consume the iterator until it finds a match (or reaches the end):
```python
>>> a = map(int, range(5))
>>> 4 in a
True
>>> 4 in a
False
```
There's actually a `partition` utility in odoo.tools.misc, so you could just use that. Or hand-roll it with a regular `for` loop and a conditional if you fear the performance impact of a function call.